pkg/apis/resource/types.go

```diff
@@ -0,0 +1,449 @@
+// uses one or more unallocated ResourceClaims with delayed allocation.
+//
+// This is an alpha type and requires enabling the DynamicResourceAllocation
+// feature gate.
+type PodScheduling struct {
+	metav1.TypeMeta
+
+	// The name must be the same as the corresponding Pod.  That Pod must
+	// be listed as owner in OwnerReferences to ensure that the
+	// PodScheduling object gets deleted when no longer needed. Normally
+	// the scheduler will delete it.
+	//
+	// Drivers must ignore PodScheduling objects where the owning pod
+	// already got deleted because such objects are orphaned and will be
+	// removed soon.
+	metav1.ObjectMeta
+
+	// Spec is set and updated by the scheduler.
+	Spec PodSchedulingSpec
+
+	// Status is updated by resource drivers.
+	Status PodSchedulingStatus
+}
+
+// PodSchedulingSpec contains the request for information about resources
+// required by a pod and eventually communicates the decision of the scheduler
+// to move ahead with pod scheduling for a specific node.
+type PodSchedulingSpec struct {
+	// When allocation is delayed, the scheduler must set the node for
+	// which it wants the resource(s) to be allocated before the driver(s)
+	// start with allocation.
+	//
+	// The driver must ensure that the allocated resource is available on
+	// this node or update PodSchedulingStatus.UnsuitableNodes to
+	// indicate where allocation might succeed.
+	//
+	// When allocation succeeds, drivers should immediately add the pod to
+	// the ResourceClaimStatus.ReservedFor field together with setting
+	// ResourceClaimStatus.Allocated. This optimization may save scheduling
+	// attempts and roundtrips through the API server because the scheduler
+	// does not need to reserve the claim for the pod itself.
+	//
+	// The selected node may change over time, for example when the initial
+	// choice turns out to be unsuitable after all. Drivers must not
+	// reallocate for a different node when they see such a change because
+	// it would lead to race conditions. Instead, the scheduler will
+	// trigger deallocation of specific claims as needed through the
+	// ResourceClaimStatus.DeallocationRequested field.
+	SelectedNode string
+
+	// When allocation is delayed, and the scheduler needs to decide on
+	// which node a Pod should run, it will ask the driver(s) on which
+	// nodes the resource might be made available. To trigger that check,
+	// the scheduler provides the names of nodes which might be suitable
+	// for the Pod. Will be updated periodically until all resources are
+	// allocated.
+	//
+	// The ResourceClass.SuiteableNodes node selector can be used to filter
+	// out nodes based on labels. This prevents adding nodes here that the
+	// driver then would need to reject through UnsuitableNodes.
+	//
+	// The size of this field is limited to 128. This is large enough for
+	// many clusters. Larger clusters may need more attempts to find a node
+	// that suits all pending resources.
```

Comment:
This should be something like:
```golang
	// For `WaitForFirstConsumer` allocations: a subset of the candidate notes where this Pod
	// might be scheduled. The API server may limit the number of entries.
	//
	// For `Immediate` allocations, ??? 
	SelectedNode string
```

but the existing comment wasn't clear enough about what this should be for the immediate allocation case for me to make a suggestion.

Also, `SuiteableNodes` contains a spelling mistake.

Reply:
See above.